### PR TITLE
Ashishonce/integration tests

### PR DESCRIPTION
**Pull request by ashishonce**
Integration test for training on aml compute. 
Will add similar test for aks compute as well, after 1st review. 

We do not need resource clean-up in this case as we want to have persistent workspace and computes for this test. 
first time the test runs, it will create the resources and hence forth those resources will be used. saving the time for creation of resource. 

We do not wait for training completion here as it is an optional feature, also it will create a long delay in test runs.

### CODE REVIEW

File: .cloud/.azure/test/train_withYamlRunConfig.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "experiment_name": "aml-run-intTest",
+    "tags": {"traing1": "t1"},
+    "wait_for_completion": false,
+    "runconfig_python_function_name": "main",
+    "runconfig_yaml_file": "test_data/train_with_yamlConfig/run_config.yml",
+    "pipeline_yaml_file": "<your-pipeline-yaml-file>",
+    "pipeline_publish": false,
+    "pipeline_name": "<your-pipeline-name>",
```
Comment (awmatheson) [inline .cloud/.azure/test/train_withYamlRunConfig.json:9]: Same comment - unused fields.

File: .cloud/.azure/test/test_aks_compute.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "name": "aks-intTest",
+    "compute_type": "akscluster",
+    "agent_count": 3,
+    "vm_size": "Standard_D3_v2",
+    "location": "<your-location>",
+    "service_cidr": "<your-service-cidr>",
+    "dns_service_ip": "<your-dns-service-ip>",
+    "docker_bridge_cidr": "<your-docker-bridge-cidr>",
+    "cluster_purpose": "<your-cluster-purpose>",
+    "vnet_resource_group_name": "<your-vnet-resource-group-name>",
```
Comment (awmatheson) [inline .cloud/.azure/test/test_aks_compute.json:11]: We can probably remove unused fields. Thoughts @marvinbuss

File: .cloud/.azure/test/test_aml_compute.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "aml-intTest",
+    "compute_type": "amlcluster",
+    "vm_size": "STANDARD_D2_V2",
+    "vm_priority": "<'dedicated' or 'lowpriority'>",
+    "min_nodes": 0,
+    "max_nodes": 4,
+    "idle_seconds_before_scaledown": 300,
+    "vnet_resource_group_name": "<your-vnet-resource-group-name>",
```
Comment (awmatheson) [inline .cloud/.azure/test/test_aml_compute.json:9]: Same comment - unused fields

File: .cloud/.azure/test/train_withConfigpy.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+    "experiment_name": "aml-run-intTest",
+    "tags": {"traing1": "t1"},
+    "wait_for_completion": false,
+    "runconfig_python_file": "test_data/train_with_configpy/run_config.py",
+    "runconfig_python_function_name": "main",
+    "runconfig_yaml_file": "<your-runconfig-yaml-file>",
+    "pipeline_yaml_file": "<your-pipeline-yaml-file>",
+    "pipeline_publish": false,
+    "pipeline_name": "<your-pipeline-name>",
```
Comment (awmatheson) [inline .cloud/.azure/test/train_withConfigpy.json:10]: Same comment - unused fields.

File: .github/workflows/integration.yml [hunk 5]
```diff
@@ -1,16 +1,45 @@
...
+      # send for training and do not wait.
+      - name: Train Script 2
+        id: ml_train2
+        uses: ./
         with:
           azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
-        continue-on-error: true
+          parameters_file: "test/train_withConfigpy.json"
+        
+      # No resource Clean-up is required in this case as resource provision and de-provision are not responsibilities of this action
```
Comment (awmatheson) [inline .github/workflows/integration.yml:45]: If we do not clean up in this action, will it be a separate action in this repo or somewhere else? Would it be easier to make a cleanup action?

File: .cloud/.azure/test/test_aks_compute.json (hunk 2 above)
Comment (marvinbuss) [reply .cloud/.azure/test/test_aks_compute.json:11]: Yes I agree.

File: .cloud/.azure/test/train_withYamlRunConfig.json (hunk 1 above)
Comment (ashishonce) [reply .cloud/.azure/test/train_withYamlRunConfig.json:9]: yes should remove the un-used fields

File: .github/workflows/integration.yml (hunk 5 above)
Comment (ashishonce) [reply .github/workflows/integration.yml:45]: training is not a persistent things happening, What I meant was, we don't need to delete the workspace and compute after every training. 
we can have dedicated workspace and compute for running aml-train integration tests

**Pull request merged**